### PR TITLE
feature: support compressing images when generate templates of blog…

### PR DESCRIPTION
**Pull request by liwang2019**
… posts.'

### CODE REVIEW

File: v2/util/gen-templates.js [hunk 1]
```diff
@@ -45,10 +52,50 @@ async function genSlideTemplate(lang) {
...
+
+  writeFile(`./templates/posts-slide-${lang}.tt2`, $Template.html() + '\n');
+};
+
+Promise.all([
+  genSlideTemplate('en'),
+  genSlideTemplate('cn'),
+]).then(() => {
+  for(pic of pics) {
+    (async() => {
```
Comment (meathill) [inline v2/util/gen-templates.js:70]: This loop will create several Async Functions, run independently with each other. That means these lines has no real meaning.

Comment (liwang2019) [reply v2/util/gen-templates.js]: changed.

File: v2/util/posts-slide.pug [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+            a.article-item(href=`${val.href}?src=org` target="_blank")
+              .blog-img(data-pic=`${val.pic}`)
+                img.article-img(src=`../images${val.pic}` loading="lazy")
+              .article-text
+                p.blog-title= val.title
+      .swiper-pagination
+    .swiper-button-next
+    .swiper-button-prev
+
+link(rel="stylesheet" href="/css/swiper.min.css")
```
Comment (meathill) [inline v2/util/posts-slide.pug:22]: Using assets on our CDN will be better.

File: v2/util/gen-templates.js [hunk 3]
```diff
@@ -1,54 +1,81 @@
...
-genSlideTemplate('cn');
+async function compressImg(filePath) {
+  return sharp(filePath)
+    .resize({
+      height,
+      fit: 'contain',
+      withoutEnlargement: true,
+    })
+    .toBuffer()
+    .then(data => {
```
Comment (meathill) [inline v2/util/gen-templates.js:75]: Again, just use `await` in Async functions.

File: v2/util/posts-slide.pug (hunk 2 above)
Comment (liwang2019) [reply v2/util/posts-slide.pug:22]: changed.

File: v2/util/gen-templates.js (hunk 3 above)
Comment (liwang2019) [reply v2/util/gen-templates.js]: changed.

**Review by meathill — APPROVED** [review]
LGTM.

**Pull request merged**